src/main/java/io/github/mfvanek/pg/connection/HighAvailabilityPgConnectionImpl.java

```diff
@@ -53,12 +67,49 @@ public Set<PgConnection> getConnectionsToAllHostsInCluster() {
 
     @Nonnull
     public static HighAvailabilityPgConnection of(@Nonnull final PgConnection connectionToPrimary) {
-        return new HighAvailabilityPgConnectionImpl(connectionToPrimary, Collections.singleton(connectionToPrimary));
+        final PrimaryHostDeterminer primaryHostDeterminer = new PrimaryHostDeterminerImpl();
+        return new HighAvailabilityPgConnectionImpl(connectionToPrimary, Collections.singleton(connectionToPrimary), primaryHostDeterminer);
     }
 
     @Nonnull
     public static HighAvailabilityPgConnection of(@Nonnull final PgConnection connectionToPrimary,
                                                   @Nonnull final Collection<PgConnection> connectionsToAllHostsInCluster) {
-        return new HighAvailabilityPgConnectionImpl(connectionToPrimary, connectionsToAllHostsInCluster);
+        final PrimaryHostDeterminer primaryHostDeterminer = new PrimaryHostDeterminerImpl();
+        final HighAvailabilityPgConnectionImpl highAvailabilityPgConnection = new HighAvailabilityPgConnectionImpl(connectionToPrimary, connectionsToAllHostsInCluster, primaryHostDeterminer);
+        highAvailabilityPgConnection.startPrimaryUpdater(DEFAULT_DELAY_SECONDS);
+        return highAvailabilityPgConnection;
+    }
+
+    @Nonnull
+    public static HighAvailabilityPgConnection of(@Nonnull final PgConnection connectionToPrimary,
+                                                  @Nonnull final Collection<PgConnection> connectionsToAllHostsInCluster,
+                                                  @Nonnull final Integer delaySeconds) {
+        final PrimaryHostDeterminer primaryHostDeterminer = new PrimaryHostDeterminerImpl();
+        final HighAvailabilityPgConnectionImpl highAvailabilityPgConnection = new HighAvailabilityPgConnectionImpl(connectionToPrimary, connectionsToAllHostsInCluster, primaryHostDeterminer);
+        highAvailabilityPgConnection.startPrimaryUpdater(delaySeconds);
+        return highAvailabilityPgConnection;
+    }
+
+    private void startPrimaryUpdater(@Nonnull final Integer delaySeconds) {
+        Objects.requireNonNull(delaySeconds, "delaySeconds");
+        if (this.getConnectionsToAllHostsInCluster().size() > 1) {
+            executorService.scheduleWithFixedDelay(this::updateConnectionToPrimary, delaySeconds, delaySeconds, TimeUnit.SECONDS);
```

Comment:
We need to cover this logic with proper test to ensure that delaySeconds parameter works as expected.
Try to decrease delaySeconds and check amount of interactions via Mockito.spy